src/main/scala/is/hail/stats/DownsampleCombiner.scala

```diff
@@ -0,0 +1,110 @@
+package is.hail.stats
+
+import org.apache.spark.sql.Row
+import is.hail.utils._
+
+class DownsampleCombiner(nDivisions: Int) extends Serializable {
+  type BinType = (Int, Int)
+  type PointType = (Double, Double, Seq[String])
+
+  var binToPoint: Map[BinType, PointType] = _
+  var initPoint: PointType = _
+
+  var left: Double = 0.0
+  var right: Double = 0.0
+  var bottom: Double = 0.0
+  var top: Double = 0.0
+
+  def toBinCoords(x: Double, y: Double, left: Double, right: Double, bottom: Double, top: Double, nDivisions: Int): (Int, Int) = {
```

Comment:
It doesn't look like this needs left, right, bottom, top, nDivisions since those are all directly accessible from here.